pkg/query/union_test.go

```diff
@@ -101,12 +101,12 @@ func TestUnionIterator(t *testing.T) {
 		documentAccess := NewDocumentAccessFixedIterator()
 		union.addSubIterator(documentAccess)
 
-		relSeq, err := ctx.Check(union, []Object{}, NewObject("user", "alice").WithEllipses())
+		pathSeq, err := ctx.Check(union, []Object{}, NewObject("user", "alice").WithEllipses())
 		require.NoError(err)
 
-		rels, err := CollectAll(relSeq)
+		paths, err := CollectAll(pathSeq)
 		require.NoError(err)
-		require.Empty(rels, "empty resource list should return no results")
+		require.Empty(paths, "empty resource list should return no results")
 	})
 
 	t.Run("Check_EarlyTermination", func(t *testing.T) {
```

Comment:
unrelated to this PR. I don't understand this test, where is the early termination happening inside the union iterator? there can't be early termination in a union, right? you need to yield them all